### PR TITLE
DB: Deprecate - get_builder

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: includes/db.php [hunk 1]
```diff
@@ -86,6 +87,8 @@ class DB {
 	 * @return array Editor data.
 	 */
 	public function get_builder( $post_id, $status = self::STATUS_PUBLISH ) {
+		_deprecated_function( __METHOD__, '3.1.0', 'Plugin::$instance->documents->get( $post_id )->get_elements_raw_data( null, true )' );
```
Comment (matipojo) [inline includes/db.php:90]: Ask @iNewLegend how to deprecate in the new way + add his PR as a dependency.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/db.php [hunk 2]
```diff
@@ -76,16 +76,26 @@ class DB {
...
 	public function get_builder( $post_id, $status = self::STATUS_PUBLISH ) {
+		Plugin::$instance->modules_manager
+			->get_modules( 'dev-tools' )
+			->deprecation
+			->deprecated_function(
+				__METHOD__,
+				'3.1.0',
+				'Plugin::$instance->documents->get( $post_id )->get_elements_raw_data( null, true )'
+			);
+
```
Comment (matipojo) [inline includes/db.php:98]: It needs to return `Plugin::$instance->documents->get( $post_id )->get_elements_raw_data( null, true )`
But, what will happen with the `status` args? (for 3rd users that uses it)

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/template-library/sources/local.php [hunk 3]
```diff
@@ -622,20 +622,22 @@ public function get_item( $template_id ) {
...
-		} else {
-			$document = Plugin::$instance->documents->get( $template_id );
-			$content = $document ? $document->get_elements_data() : [];
-		}
+		$document = Plugin::$instance->documents->get( $template_id );
+		$content = [];
 
-		if ( ! empty( $content ) ) {
-			$content = $this->replace_elements_ids( $content );
+		if ( $document ) {
```
Comment (matipojo) [inline includes/template-library/sources/local.php:630]: Tests?

- $args['display'] - is empty
- $args['display'] - not empty
- document with content
- document without content

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/includes/template-library/test-manager-local.php [hunk 4]
```diff
@@ -141,4 +142,67 @@ public function test_should_return_data_from_get_template_data() {
 
 		$this->assertEquals( $ret, [ 'content' => [] ] );
 	}
+
+	public function test_get_data__document_without_data() {
+		$is_edit_mode = Plugin::$instance->editor->is_edit_mode();
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/template-library/test-manager-local.php:147]: // AAA for all tests.

File: tests/phpunit/elementor/includes/template-library/test-manager-local.php [hunk 5]
```diff
@@ -141,4 +142,67 @@ public function test_should_return_data_from_get_template_data() {
...
+				'_elementor_data' => [],
+			],
+		] );
+
+		$result = Plugin::$instance->templates_manager->get_source( 'local' )->get_data( [
+			'template_id' => $post->ID,
+		] );
+
+		$this->assertArrayHasKey( 'content', $result );
+		$this->assertEmpty( $result['content'] );
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/template-library/test-manager-local.php:162]: Assers it's an array.

File: tests/phpunit/elementor/includes/template-library/test-manager-local.php [hunk 6]
```diff
@@ -141,4 +142,67 @@ public function test_should_return_data_from_get_template_data() {
...
+			[ 'settings', 'elements', 'isInner' ],
+			$result['content'][0],
+			'should NOT have those keys if display is not exists in the args'
+		);
+		$this->assertEquals( 'section', $result['content'][0]['elType'] );
+		$this->assertNotEquals( 'test', $result['content'][0]['id'] );
+
+		$result = Plugin::$instance->templates_manager->get_source( 'local' )->get_data( [
+			'template_id' => $post->ID,
+			'display' => 'something',
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/template-library/test-manager-local.php:195]:
```suggestion
			'display' => true,
```

File: tests/phpunit/elementor/includes/template-library/test-manager-local.php [hunk 7]
```diff
@@ -141,4 +142,67 @@ public function test_should_return_data_from_get_template_data() {
...
+		] );
+
+		$this->assertArrayHasKey( 'content', $result );
+		$this->assertArrayHaveKeys( [ 'id', 'elType' ], $result['content'][0] );
+		$this->assertArrayHaveKeys(
+			[ 'settings', 'elements', 'isInner' ],
+			$result['content'][0],
+			'SHOULD have those keys if display is not exists in the args'
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/template-library/test-manager-local.php:203]:
```suggestion
			'SHOULD have those keys if display is exists in the args'
```

File: tests/phpunit/elementor/includes/template-library/test-manager-local.php [hunk 8]
```diff
@@ -141,4 +142,67 @@ public function test_should_return_data_from_get_template_data() {
...
+
+		$this->assertArrayHasKey( 'content', $result );
+		$this->assertArrayHaveKeys( [ 'id', 'elType' ], $result['content'][0] );
+		$this->assertArrayNotHaveKeys(
+			[ 'settings', 'elements', 'isInner' ],
+			$result['content'][0],
+			'should NOT have those keys if display is not exists in the args'
+		);
+		$this->assertEquals( 'section', $result['content'][0]['elType'] );
+		$this->assertNotEquals( 'test', $result['content'][0]['id'] );
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/template-library/test-manager-local.php:191]: Why not?

File: tests/phpunit/elementor/includes/template-library/test-manager-local.php [hunk 9]
```diff
@@ -141,4 +142,67 @@ public function test_should_return_data_from_get_template_data() {
...
+
+		$this->assertArrayHasKey( 'content', $result );
+		$this->assertArrayHaveKeys( [ 'id', 'elType' ], $result['content'][0] );
+		$this->assertArrayHaveKeys(
+			[ 'settings', 'elements', 'isInner' ],
+			$result['content'][0],
+			'SHOULD have those keys if display is not exists in the args'
+		);
+		$this->assertEquals( 'section', $result['content'][0]['elType'] );
+		$this->assertNotEquals( 'test', $result['content'][0]['id'] );
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/template-library/test-manager-local.php:206]: Why not?

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

File: tests/phpunit/elementor/includes/template-library/test-manager-local.php (hunk 5 above)
Comment (Nevoss) [reply tests/phpunit/elementor/includes/template-library/test-manager-local.php:166]: `assertIsArray()` is not exists in php 5.6. so I used `assertIsTrue(is_array($result['content']))`

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

**Review by kobizz — DISMISSED** [review]

**Review by KingYes — DISMISSED** [review]

**Pull request merged**